concordium-sdk/src/main/java/com/concordium/sdk/transactions/CredentialRegistrationId.java

```diff
@@ -0,0 +1,49 @@
+package com.concordium.sdk.transactions;
+
+import lombok.*;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+
+import java.util.Objects;
+
+/**
+ * A credential registration id
+ */
+@ToString
+@EqualsAndHashCode
+public class CredentialRegistrationId {
+    //the length of `CredentialRegistrationId` bytes
+    private static final int LENGTH = 48;
+    @Getter
+    private final byte[] regId;
+
+    private CredentialRegistrationId(byte[] regId) {
+        this.regId = regId;
+    }
+
+    /**
+     * Create a {@link CredentialRegistrationId} from the hex encoded registration id.
+     */
+    public static CredentialRegistrationId from(String encoded) {
+        try {
+            val regId = Hex.decodeHex(encoded);
+            return CredentialRegistrationId.fromBytes(regId);
+        } catch (DecoderException e) {
+            throw new IllegalArgumentException("CredentialRegistrationId is not a valid HEX encoding", e);
+        }
+    }
+
+    /**
+     * Create a {@link CredentialRegistrationId} from raw bytes
+     *
+     * @param bytes the credential registration id bytes
+     * @return a {@link CredentialRegistrationId}
+     */
+    public static CredentialRegistrationId fromBytes(byte[] bytes) {
+        if (bytes.length != LENGTH) {
```

Comment:
In principle a credential registration ID is a group element in this G1 curve of the BLS pairing.

So not every series of bytes is a valid credential registration ID (in fact I think the probability that a random 48 bytes is a valid group element is somewhere around (1/2^17). I think this is fine for this version, but perhaps some documentation or "note for maintainers" could be added since in the future we might want to check validity.

Reply:
Yes that is correct. I will add some documentation covering this subject. I don't think we need the implementation for verifying whether it's a group element of the group in question right now. 

So yes I agree :)